lib/ansible/module_utils/facts/hardware/hpux.py

```
@@ -0,0 +1,161 @@
+                # if machinfo return cores strings release B.11.31 > 1204
+                rc, out, err = self.module.run_command("/usr/contrib/bin/machinfo | grep core | wc -l", use_unsafe_shell=True)
+                if out.strip() == '0':
+                    rc, out, err = self.module.run_command("/usr/contrib/bin/machinfo | grep Intel", use_unsafe_shell=True)
+                    cpu_facts['processor_count'] = int(out.strip().split(" ")[0])
+                    # If hyperthreading is active divide cores by 2
+                    rc, out, err = self.module.run_command("/usr/sbin/psrset | grep LCPU", use_unsafe_shell=True)
+                    data = re.sub(' +', ' ', out).strip().split(' ')
+                    if len(data) == 1:
+                        hyperthreading = 'OFF'
+                    else:
+                        hyperthreading = data[1]
```

Ruff (0.12.2)

Use ternary operator `hyperthreading = 'OFF' if len(data) == 1 else data[1]` instead of `if`-`else`-block

Replace `if`-`else`-block with `hyperthreading = 'OFF' if len(data) == 1 else data[1]`

(SIM108)